src/main/ipv-stub/ipv-token.ts

```diff
@@ -1,30 +1,129 @@
 import {
   APIGatewayProxyEvent,
+  APIGatewayProxyEventHeaders,
   APIGatewayProxyResult,
   Handler,
 } from "aws-lambda";
+import querystring from "node:querystring";
+import { importSPKI, jwtVerify } from "jose";
+import { ACCESS_TOKEN, AUTH_CODE } from "./data/ipv-dummy-constants";
+import {
+  CodedError,
+  handleErrors,
+  methodNotAllowedError,
+  successfulJsonResult,
+} from "./helper/result-helper";
 
 export const handler: Handler = async (
   event: APIGatewayProxyEvent
 ): Promise<APIGatewayProxyResult> => {
-  switch (event.httpMethod) {
-    case "GET":
-      return get(event);
-    default:
-      return {
-        statusCode: 405,
-        body: JSON.stringify({
-          message: "Method not allowed",
-        }),
-      };
-  }
+  return handleErrors(async () => {
+    switch (event.httpMethod) {
+      case "POST":
+        return await post(event);
+      default:
+        throw methodNotAllowedError(event.httpMethod);
+    }
+  });
 };
 
-function get(event: APIGatewayProxyEvent): APIGatewayProxyResult {
-  return {
-    statusCode: 200,
-    body: JSON.stringify({
-      message: `Reached the ${event.httpMethod} endpoint`,
-    }),
-  };
+async function post(
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> {
+  validateHeadersOrThrow(event.headers);
+  const body = getValidBodyOrThrow(event.body);
+  const clientAssertionJwt = body["client_assertion"] as string;
+
+  const publicKey = await importSPKI(
+    process.env.IPV_TOKEN_PUBLIC_SIGNING_KEY as string,
+    "ES256"
+  );
+
+  try {
+    await jwtVerify(clientAssertionJwt, publicKey, {
+      audience: process.env.AUTH_FRONTEND_BASE_URL,
```

Comment:
How does this get set? And why is it auth?

Reply:
Changed to IPV_AUDIENCE. Looks like it's actually undefined in dev.

Reply:
It's undefined in dev and build because those envs aren't currently configured for identity journeys (see `IpvExists` condition in the template). However once this stub is ready we will enable identity journeys in dev and build, so this should be set to `https://identity.<environment>.account.gov.uk` in non-prod and `https://identity.account.gov.uk` in prod.